### PR TITLE
Pm manager outliers

### PR DESCRIPTION
**Pull request by kevinverhoff**
# Pull Request

## Summary & Motivation
adding observer_employee_number:  I want to add observer employee number to the observations view in order to link observations to other information about observers.

creating score_averages: This new view will allow analysis on managers' score averages on ETR and S&O, namely outlier analysis.

[//]: # (When merged, this pull request will...)
-add the primary key to observation_details for observers
-create a manager score averages table that we can use for analysis projects

## Self-review

- [x] Update **due date**, **assignee**, and **priority** on our [TEAMster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)

    _If this is a same-day request, please flag that in Slack_

- [x] <kbd>Format</kbd> has been run on all modified files

## Troubleshooting
- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206329661627627

### CODE REVIEW

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__schoolmint_grow_observation_details.sql [hunk 1]
```diff
@@ -56,88 +63,91 @@ with
...
             /* remove self coaching observations*/
             and case
                 when
-                    rubric_name = 'Coaching Tool: Coach ETR and Reflection'
-                    and teacher_id = observer_id
+                    o.rubric_name = 'Coaching Tool: Coach ETR and Reflection'
+                    and o.teacher_id = observer_id
                 then false
                 else true
             end
```
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__schoolmint_grow_observation_details.sql:153]:
```
and not (o.rubric_name = 'Coaching Tool: Coach ETR and Reflection' and o.teacher_id = o.observer_id)
```

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__schoolmint_grow_observation_details.sql [hunk 2]
```diff
@@ -487,6 +500,7 @@ with
             od.created,
             od.observed_at,
             od.observer_name,
+            cast(od.observer_employee_number as int64) as observer_employee_number,
```
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__schoolmint_grow_observation_details.sql:503]: - [ ] move to original cte
- [ ] use `safe_cast`
- [ ] `int` should be sufficient

File: src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql
Comment (cbini) [inline src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql:1]: this isn't a staging view, is this for tableau?

File: src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql [hunk 3]
```diff
@@ -0,0 +1,167 @@
+with
+    measures_long as (
+        select
+            observer_employee_number,
+            academic_year,
+            form_term,
+            row_score_value,
+            -- trunk-ignore-begin(sqlfluff)
+            case
```
Comment (cbini) [inline src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql:9]: are these codes available from schoolmint?

File: src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                    = 'This teammate seeks feedback and takes it seriously.'
+                then 'SO7'
+                when
+                    measurement_name
+                    = 'This teammate supports the collective work of the team and meets team commitments.'
+                then 'SO8'
+                else concat('ETR', left(measurement_name, 2))
+            end as measurement_code,
+        -- trunk-ignore-end(sqlfluff)
+        from {{ ref("rpt_tableau__schoolmint_grow_observation_details") }}
```
Comment (cbini) [inline src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql:45]: use the schoolmint staging tables or refactor the data you need from the tableau view to a intermediate view that both reporting views can use

File: src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+            )
+    ),
+
+    manager_overall as (
+        select
+            observer_employee_number,
+            academic_year,
+            form_term,
+            avg(overall_score) as overall_score,
+        from {{ ref("rpt_tableau__schoolmint_grow_observation_details") }}
```
Comment (cbini) [inline src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql:123]: same as above

**Review by cbini — CHANGES_REQUESTED** [review]

File: src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql [hunk 6]
```diff
@@ -0,0 +1,167 @@
+with
+    measures_long as (
+        select
+            observer_employee_number,
+            academic_year,
+            form_term,
+            row_score_value,
+            -- trunk-ignore-begin(sqlfluff)
```
Comment (cbini) [inline src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql:8]: specify the linter code that you're ignoring, this ignores all sqlfluff checks

**Review by cbini — CHANGES_REQUESTED** [review]

**Comment by cbini** [comment]
closes #478

File: src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql:1]: I'm not sure what to call it. it's ultimately going to be for the little python tool we're scheduled to talk about. Should we create a new folder in extracts?

Comment (cbini) [reply src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql:1]: yeah that makes more sense

File: src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql (hunk 4 above)
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql:45]: done

File: src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql (hunk 3 above)
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql:9]: no, but we set up a regex for it

File: src/dbt/kipptaf/models/schoolmint/grow/intermediate/int_schoolmint_grow__observation_details.sql
Comment (cbini) [inline src/dbt/kipptaf/models/schoolmint/grow/intermediate/int_schoolmint_grow__observation_details.sql:1]: can you put this under `performance_management`?

File: src/dbt/kipptaf/models/schoolmint/grow/intermediate/int_schoolmint_grow__observation_details.sql [hunk 7]
```diff
@@ -0,0 +1,540 @@
...
+        -- sr.report_to_mail,
+        -- sr.sam_account_name,
+        -- sr.report_to_sam_account_name,
+        from {{ ref("stg_schoolmint_grow__users__roles") }} as ur
+        left join {{ ref("stg_schoolmint_grow__users") }} as u on ur.user_id = u.user_id
+        left join
+            {{ ref("stg_reporting__terms") }} as rt on ur.role_name = rt.grade_band
+        -- left join
+        -- {{ ref("base_people__staff_roster") }} as sr
+        -- on u.internal_id = safe_cast(sr.employee_number as string)
```
Comment (cbini) [inline src/dbt/kipptaf/models/schoolmint/grow/intermediate/int_schoolmint_grow__observation_details.sql:38]: remove commented out lines

File: src/dbt/kipptaf/models/extracts/python_tasks/rpt_python_tasks__manager_pm_averages.sql
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/python_tasks/rpt_python_tasks__manager_pm_averages.sql:1]: `extracts/performance_management`?

**Review by cbini — CHANGES_REQUESTED** [review]

File: src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql:1]: done

File: src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql (hunk 5 above)
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/people/staging/stg_people__manager_pm_score_averages.sql]: done

File: src/dbt/kipptaf/models/schoolmint/grow/intermediate/int_schoolmint_grow__observation_details.sql (hunk 7 above)
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/schoolmint/grow/intermediate/int_schoolmint_grow__observation_details.sql]: done

File: src/dbt/kipptaf/models/schoolmint/grow/intermediate/int_schoolmint_grow__observation_details.sql
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/schoolmint/grow/intermediate/int_schoolmint_grow__observation_details.sql:1]: yep. done

File: src/dbt/kipptaf/models/extracts/python_tasks/rpt_python_tasks__manager_pm_averages.sql
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/extracts/python_tasks/rpt_python_tasks__manager_pm_averages.sql:1]: checking in on slack... extracts folders seem to be named after their destinations, which is why I named it this way. I'm open to whatever - me da igual

Comment (kevinverhoff) [reply src/dbt/kipptaf/models/extracts/python_tasks/rpt_python_tasks__manager_pm_averages.sql:1]: per slack, switching to 'python'

**Review by cbini — APPROVED** [review]

**Pull request merged**